skglm/solvers/base.py

```diff
@@ -1,8 +1,26 @@
-from abc import abstractmethod
+from abc import abstractmethod, ABC
+from skglm.utils.validation import check_obj_solver_attr
 
 
-class BaseSolver():
-    """Base class for solvers."""
+class BaseSolver(ABC):
+    """Base class for solvers.
+
+    Attributes
+    ----------
+    _datafit_required_attr : list
```

Comment:
I think we can make these public @BadrMoufad ? Maybe I'm missing a specific reason

Reply:
also typo missing "that must BE" here and below

Reply:
> I think we can make these public @BadrMOUFAD ? Maybe I'm missing a specific reason

I think these two attributes should be read only.

While there is a way to make attributes read only, namely using [the property decorator](https://docs.python.org/3/library/functions.html#property), I believe it adds two much complexity to the code and hence doesn’t serve our goal to make components implementation user-friendly.

I opted for the “start with underscore” naming convention to make variables private to signal to the user that these are attributes to not mess up with.